### DISCUSSION TITLE
How to change names of labels in the legend?

### DISCUSSION THREAD

**Discussion by TRo202** [facontidavide/PlotJuggler#779] [Q&A] [answered]
Hi

I'm new to Plotjuggler, and I couldn't find a way to change names of labels in the legend.
Is there anyone who knows how?
I'll really appreciate it.

**Comment by Bartimaeus-** [comment, marked as answer]
Presently there are two ways this can be done in Plotjuggler:

1) Apply a transformation (i.e. apply a scaling factor of 1) to the curve (only changes the name for that plot)

2) Create a new custom series with the name you want that simply returns the values of the original function (creates a new series with a different name)

See here for a video clip of the transformation naming method: https://github.com/facontidavide/PlotJuggler/issues/656

**Reply by TRo202** [reply]
Oh thanks for your answer! 

And oh I wasn't the only one who knew the first way! I forgot to add that on my question, cuz I was actually wondering if there was any "proper" way to do it.

Hopefully, they would add a "proper" way to do so in future update

**Comment by facontidavide** [comment]
yes, that is the dirty trick. otherwise, there is no support for that
